uaclient/tests/test_cli.py

```diff
@@ -15,6 +18,78 @@
 from uaclient.testing.fakes import FakeConfig
 
 
+BIG_DESC = "123456789 " * 7 + "next line"
+BIG_URL = "http://" + "adsf" * 10
+
+
+class TestCLIParser:
+    @mock.patch("uaclient.cli.entitlements")
+    def test_help_descr_and_url_is_wrapped_at_eighty_chars(
+        self, m_entitlements, tmpdir
+    ):
+        """Help lines are wrapped at 80 chars"""
+        help_file = tmpdir.join("log_file").strpath
+
+        def cls_mock_factory(desc, url):
+            return mock.Mock(description=desc, help_doc_url=url)
+
+        m_entitlements.ENTITLEMENT_CLASS_BY_NAME = {
+            "test": cls_mock_factory(BIG_DESC, BIG_URL)
+        }
+
+        parser = get_parser()
+        with open(help_file, "w") as stream:
+            parser.print_help(file=stream)
+        output = util.load_file(help_file)
+        lines = [
+            " - test: " + " ".join(["123456789"] * 7),
+            "   next line ({url})".format(url=BIG_URL),
+        ]
+        assert "\n".join(lines) in output
+
+    def test_help_sourced_dynamically_from_each_entitlement(self, tmpdir):
+        """Help output is sourced from entitlement name and description."""
+        help_file = tmpdir.join("log_file").strpath
+        parser = get_parser()
+        cc_eal = ENTITLEMENT_CLASS_BY_NAME["cc-eal"]
+        cis_audit = ENTITLEMENT_CLASS_BY_NAME["cis-audit"]
+        esm = ENTITLEMENT_CLASS_BY_NAME["esm-infra"]
+        fips = ENTITLEMENT_CLASS_BY_NAME["fips"]
+        fips_updates = ENTITLEMENT_CLASS_BY_NAME["fips-updates"]
+        livepatch = ENTITLEMENT_CLASS_BY_NAME["livepatch"]
+        fmt_params = {
+            "cc_desc": cc_eal.description,
+            "cc_url": cc_eal.help_doc_url,
+            "cis_desc": cis_audit.description,
+            "cis_url": cis_audit.help_doc_url,
+            "esm_desc": esm.description,
+            "esm_url": esm.help_doc_url,
+            "fips_desc": fips.description,
+            "fips_url": fips.help_doc_url,
+            "fips_updates_desc": fips_updates.description,
+            "fips_updates_url": fips_updates.help_doc_url,
+            "livepatch_desc": livepatch.description,
+            "livepatch_url": livepatch.help_doc_url,
+        }
+        expected_output = """
+ - cc-eal: {cc_desc}
+   ({cc_url})
+ - cis-audit: {cis_desc}
+   ({cis_url})
+ - esm-infra: {esm_desc} ({esm_url})
+ - fips: {fips_desc} ({fips_url})
+ - fips-updates: {fips_updates_desc}
+ - livepatch: {livepatch_desc} ({livepatch_url})
+
+""".format(
+            **fmt_params
+        )
+        with open(help_file, "w") as stream:
+            parser.print_help(file=stream)
```

Comment:
Could we just use an in-memory file-like object here, to remove the need for the temporary directory? (`io.StringIO`, perhaps?)

(This also applies to the above test case, of course.)

Reply:
done.